rust/core-lib/src/editor.rs

```diff
@@ -693,20 +758,15 @@ impl<W: Write + Send + 'static> Editor<W> {
     }
 
     fn do_cut(&mut self) -> Value {
-        let min = self.view.sel_min();
-        if min != self.view.sel_max() {
-            let val = self.text.slice_to_string(min, self.view.sel_max());
-            let del_interval = Interval::new_closed_open(min, self.view.sel_max());
-            self.add_simple_edit(del_interval, Rope::from(""));
-            Value::String(val)
-        } else {
-            Value::Null
-        }
+        let result = self.do_copy();
+        // This copy is just to make the borrow checker happy, could be optimized.
+        let deletions = self.view.sel_regions().to_vec();
+        self.delete_sel_regions(&deletions);
+        result
     }
 
-    fn do_copy(&mut self) -> Value {
-        if self.view.sel_start != self.view.sel_end {
-            let val = self.text.slice_to_string(self.view.sel_min(), self.view.sel_max());
+    fn do_copy(&self) -> Value {
+        if let Some(val) = self.extract_sel_regions(self.view.sel_regions()) {
```

Comment:
I don't love how this handles paste after multi-sel copy. It feels like, in that instance, the individual regions should be pasted, at least if the number of regions is equal.

It feels like there should be some intermediate representation, that only becomes a string when paste context is known?